builtin/credential/approle/path_role_test.go

```diff
@@ -1843,6 +1844,23 @@ func createRole(t *testing.T, b *backend, s logical.Storage, roleName, policies
 // TestAppRole_TokenutilUpgrade ensures that when we read values out that are
 // values with upgrade logic we see the correct struct entries populated
 func TestAppRole_TokenutilUpgrade(t *testing.T) {
+	tests := []struct {
```

Comment:
Would be good to also have the case where there is no token_type key at all in the JSON, which is what we’d see coming from a pre-1.0 state.